### DISCUSSION TITLE
Allow users to change the link that the plugin gives.

### DISCUSSION THREAD

**Discussion by ZepsiZola** [GriefPrevention/GriefPrevention#2121] [Ideas]
For a while I have wanted to change the bit.ly link that GriefPrevention gives the player when they pick up their gold shovel or pick up the "How to Claim Land" book. It shouldn't be too hard to just set a config one of the `.yml` files that lets you configure the link. On top of this, I would also like to be able to change the contents of the GriefPrevention book to fit my server more specifically.

**Comment by Jikoo** [comment]
You already can. The book loads its text from messages.yml, and you don't have to include the placeholder for GP's built in video URL in any of the messages.

**Reply by ZepsiZola** [reply]
But am I able to change the "built-in video URL"? I want to make it a different URL.

**Reply by Jikoo** [reply]
Just don't include the "{0}" placeholder, put your own URL there. Without it present, GP's internal URL will not be added.

**Reply by ZepsiZola** [reply]
Don't you think it would be better if we could just configure our own placeholders instead of either having to use a link to a 9 year-old YT video or completely forgo the placeholder completely? It kinda defeats the purpose of a placeholder if, to change it, you just have to manually enter what you want in every line instead.

**Reply by Jikoo** [reply]
Not really. GP hasn't changed its claim style, and you are already free to omit its video for your own. Is copying and pasting so onerous that you need a custom placeholder for use in text you are already customizing? You'd probably end up copying and pasting the placeholder, which would if anything be worse. You're free to submit a PR if you really think it's worth someone's time to do.

I believe Robo has talked about updating the video, but as someone who dislikes videos as a format for conveying information, that's not something I personally will touch. If I had to hazard a guess, I'd bet BigScary made it that way more because GP is an opinionated protection plugin than because it should be convenient for you the user to change.

**Reply by ZepsiZola** [reply]
I will end up doing what you suggest with the placeholder and link, but I think it would just make more sense if people could simply just change the placeholder. If it is possible for the user to change the link this way, why not make it easier for them to simply just change the link at one location and be done with it? Why does it have to be harder than it should be for server owners to personalize the plugin? And surely it can't be that hard to just set a config that configures the internal placeholder.

**Reply by QarthO** [reply]
you can always adjust in the src code and rebuild the plugin urself.

**Reply by ZepsiZola** [reply]
Then I would have to do that every time the plugin updates. How about this, if I make a commit that implements the feature I want and it doesn't cause any issues, can that be merged into the main project?

**Reply by QarthO** [reply]
yup! those are called pull requests, that would be perfect
